### PR TITLE
fix: Allow connecting to a server on a URL path (#149)

### LINKED ISSUE #149: Adding URL prefix to client
I'm currently trying to use this client to write to a VictoriaMetrics instance. They say that v2 API calls work correctly (https://victoriametrics.github.io/Cluster-VictoriaMetrics.html#url-format), but every time I try to write using this client, it appears the client is ignoring the prefix I set.

For example, if I set `http://localhost:8480/insert/0:0/influx` as my serverUrl, this client appears to use `http://localhost:8480/api/v2/write` instead of `http://localhost:8480/insert/0:0/influx/api/v2/write` as I would expect.

Am I doing something wrong in the client? My actual instantiation:

```
    client := influxdb2.NewClientWithOptions(
        fmt.Sprintf("http://%v:%v/insert/0:0/influx", hostname, port),
        "",
        influxdb2.DefaultOptions().SetBatchSize(BatchSize),
    )
    writeApi := client.WriteApi("", "telegraf")
```

The error I receive:

```
Jul 09 14:34:06 <> vminsert-prod[6281]: 2020-07-09T14:34:06.206Z        warn        VictoriaMetrics/lib/httpserver/httpserver.go:204        unsupported path requested: "/api/v2/write"
```

A similar error I received using the v1 client:

```root@<>:~# journalctl -xeu victoriametrics-vminsert | grep "write/write"
Jul 08 22:42:20 <> vminsert-prod[6281]: 2020-07-08T22:42:20.815Z        warn        VictoriaMetrics/lib/httpserver/httpserver.go:204        unsupported path requested: "/insert/0:0/influx/write/write"
```

In this case, we can see it used the full path I had defined:

```
c, err := client.NewHTTPClient(client.HTTPConfig{
                Addr: fmt.Sprintf("http://%v:%v/insert/0:0/influx/write", hostname, port),
        })
```

 Which leads me to believe the client is stripping any url prefix.

### PR DESCRIPTION
**Pull request by vlastahajek**
Closes #149

Properly concatenating URL path to the final API endpoint

- [X] CHANGELOG.md updated
- [X] Rebased/mergeable
- [X] Tests pass

### CODE REVIEW

File: client_test.go [hunk 1]
```diff
@@ -56,11 +85,27 @@ func TestServerError429(t *testing.T) {
...
+	err := c.WriteApiBlocking("o", "b").WriteRecord(context.Background(), "a,a=a a=1i")
+	require.Nil(t, err)
+}
+
 func TestServerErrorNonJSON(t *testing.T) {
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		time.Sleep(100 * time.Millisecond)
 		w.WriteHeader(http.StatusInternalServerError)
-		w.Write([]byte(`internal server error`))
+		_, _ = w.Write([]byte(`internal server error`))
```
Comment (sranka) [inline client_test.go:108]: https://staticcheck.io/docs/checks#S1005

File: client_test.go [hunk 2]
```diff
@@ -79,7 +124,7 @@ func TestServerErrorInflux1_8(t *testing.T) {
 		w.Header().Set("X-Influxdb-Error", "bruh moment")
 		w.Header().Set("Content-Type", "application/json")
 		w.WriteHeader(404)
-		w.Write([]byte(`{"error": "bruh moment"}`))
+		_, _ = w.Write([]byte(`{"error": "bruh moment"}`))
```
Comment (sranka) [inline client_test.go:127]: https://staticcheck.io/docs/checks#S1005

File: client_test.go [hunk 3]
```diff
@@ -42,7 +71,7 @@ func TestServerError429(t *testing.T) {
 		w.Header().Set("Retry-After", "1")
 		w.Header().Set("Content-Type", "application/json")
 		w.WriteHeader(http.StatusTooManyRequests)
-		w.Write([]byte(`{"code":"too many requests", "message":"exceeded rate limit"}`))
+		_, _ = w.Write([]byte(`{"code":"too many requests", "message":"exceeded rate limit"}`))
```
[the remark this answers is not in the document]
Comment (vlastahajek) [reply client_test.go:74]: Explicitly noting about ignoring returned val and returned error recommended by the style check in the IDE. 
It's validated by running full  profile staticcheck ('staticcheck -show-ignored -checks="all" ./...') and this case is ok

File: client_test.go [hunk 4]
```diff
@@ -6,16 +6,45 @@ package influxdb2
...
+	urls := []struct {
+		serverURL      string
+		serverAPIURL   string
+		writeURLPrefix string
+	}{
+		{"http://host:9999", "http://host:9999/api/v2/", "http://host:9999/api/v2/write"},
+		{"http://host:9999/", "http://host:9999/api/v2/", "http://host:9999/api/v2/write"},
+		{"http://host:9999/path", "http://host:9999/path/api/v2/", "http://host:9999/path/api/v2/write"},
+		{"http://host:9999/path/", "http://host:9999/path/api/v2/", "http://host:9999/path/api/v2/write"},
+		{"http://host:9999/path1/path2/path3", "http://host:9999/path1/path2/path3/api/v2/", "http://host:9999/path1/path2/path3/api/v2/write"},
```
[the remark this answers is not in the document]
Comment (vlastahajek) [reply client_test.go:31]: `/path1/path2/path3` is tested as a more complex path and they differ in the '/' at the end

**Pull request merged**